src/components/Tasks/Toolbar.tsx

```diff
@@ -20,23 +24,51 @@ import { StoreState } from '../../store';
 
 import Menu from '../../widgets/Menu';
 
+const transitionTimeout = 300;
+
+const transitionStyles = {
+  entering: { visibility: 'visible', width: '100%', overflow: 'hidden' },
+  entered: { visibility: 'visible', width: '100%' },
+  exiting: { visibility: 'visible', width: '0%', overflow: 'hidden' },
+  exited: { visibility: 'hidden', width: '0%' },
+};
+
+const useStyles = makeStyles((theme) => ({
+  button: {
+    marginRight: theme.spacing(1),
+  },
+  textField: {
+    transition: `width ${transitionTimeout}ms`,
```

Comment:
Not quite. From the `Transition` [documentation](http://reactcommunity.org/react-transition-group/transition): 
> By default the `Transition` component does not alter the behavior of the component it renders, it only tracks "enter" and "exit" states for the components. It's up to you to give meaning and effect to those states.

The component doesn't really do anything but track states. Transition styles have to be manually applied.

Reply:
So why do you need to pass the timeout to it?

Reply:
To the stye or the component?

Reply:
Why does the Transition component get a timeout if it doesn't actually do the transition? That's what I'm not getting.

Reply:
Basically all the library is doing is helping apply different inline styles at different times. The actual animation of the transition effect comes from CSS. I believe the component needs to know the timeout so it flips states at the right time. The style needs it because that determines how long the actual transition animation lasts. Hypothetically these both could be different values, but that would just cause the transitions to look weird and/or cut off.

This is just how the library does it. I know it's a little clunky.

Reply:
The `Transition` component allows you to apply different styles for 4 different phases of a transition: entering, entered, exiting, exited. It's not really doing any CSS transitions on it's own. Just transitioning between those states.

Reply:
Haha I'm having a tough time explaining it, but the [documentation](http://reactcommunity.org/react-transition-group/transition) is pretty clear.

Reply:
Cool, as long as it makes sense, I'm fine with that. Just found it odd...

Reply:
I know, they way the library works is a little strange.